src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -220,6 +221,128 @@ public override async Task<TResult> QueryWorkflowAsync<TResult>(QueryWorkflowInp
+                    {
+                        req.WaitPolicy = new() { LifecycleStage = options.WaitForStage };
+                    }
+                }
+                if (input.Args.Count > 0)
+                {
+                    req.Request.Input.Args = new Payloads();
+                    req.Request.Input.Args.Payloads_.AddRange(
+                        await Client.Options.DataConverter.ToPayloadsAsync(
+                            input.Args).ConfigureAwait(false));
+                }
+                if (input.Headers != null)
+                {
+                    req.Request.Input.Header = new();
+                    req.Request.Input.Header.Fields.Add(input.Headers);
+                    // If there is a payload codec, use it to encode the headers
+                    if (Client.Options.DataConverter.PayloadCodec is IPayloadCodec codec)
+                    {
+                        foreach (var kvp in req.Request.Input.Header.Fields)
+                        {
+                            req.Request.Input.Header.Fields[kvp.Key] =
+                                await codec.EncodeSingleAsync(kvp.Value).ConfigureAwait(false);
+                        }
+                    }
+                }
+                // Invoke
+                var resp = await Client.Connection.WorkflowService.UpdateWorkflowExecutionAsync(
+                    req, DefaultRetryOptions(input.Options?.Rpc)).ConfigureAwait(false);
+                // Build handle for result
+                return new(Client, req.Request.Meta.UpdateId, input.Id, input.RunId)
+                {
+                    // Put outcome on the handle (may be null)
+                    KnownOutcome = resp.Outcome,
+                };
+            }
+
+            /// <inheritdoc />
+            public override async Task<TResult> PollWorkflowUpdateAsync<TResult>(
+                PollWorkflowUpdateInput input)
+            {
+                var req = new PollWorkflowExecutionUpdateRequest()
+                {
+                    Namespace = Client.Options.Namespace,
+                    UpdateRef = new()
+                    {
+                        WorkflowExecution = new()
+                        {
+                            WorkflowId = input.WorkflowId,
+                            RunId = input.WorkflowRunId ?? string.Empty,
+                        },
+                        UpdateId = input.Id,
+                    },
+                    Identity = Client.Connection.Options.Identity,
+                    WaitPolicy = new() { LifecycleStage = UpdateWorkflowExecutionLifecycleStage.Completed },
+                };
+                // Create cancellation token based on timeout. This is a different timeout than RPC
+                // timeout because RPC timeout is _per call_ timeout, this is overall.
+                using (var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(
+                    input.RpcOptions?.CancellationToken ?? default))
```

Comment:
Because some calls have call-specific options which _include_ RPC options within them and some calls just accept RPC options directly because they don't need their own options object (and/or they can take parameters explicitly). The get-result call of a workflow which polls does not need a special options object so it accepts `RpcOptions` directly.

This is the same with workflows. See how starting a workflow accepts workflow options w/ RPC options inside, but getting a workflow result accepts RPC options directly. Starting/executing an update accepts update options w/ RPC options inside, but getting an update result accepts RPC options directly.